### PR TITLE
Document detailed Azure permissions to manage a Shoot

### LINKED ISSUE #26: Minimal Permissions for user credentials
From https://github.com/gardener/gardener-extensions/issues/133

> We have narrowed down the access permissions for AWS shoot clusters (potential remainder tracked in #178), but not yet for Azure, GCP and OpenStack, which this ticket is now about. We expect less success on these infrastructures as AWSes permision/policy options are very detailed. This may break the "shared account" idea on these infrastructures (Azure and GCP - OpenStack can be mitigated by programmatically creating tenants on the fly).

### PR DESCRIPTION
**Pull request by dkistner**
**How to categorize this PR?**
/area documentation
/kind enhancement
/platform azure

**What this PR does / why we need it**:
Document detailed Azure provider/service actions to manage a Shoot cluster on Azure.

**Which issue(s) this PR fixes**:
Fixes #26

**Release note**:

```doc user
A detailed list of Azure provider/service permissions/actions that are required to manage Azure Shoot clusters is now available [here](https://github.com/gardener/gardener-extension-provider-azure/blob/master/docs/azure-permissions.md).
```

/invite @kon-angelo

### CODE REVIEW

File: docs/tutorials/kubernetes-cluster-on-azure-with-gardener/kubernetes-cluster-on-azure-with-gardener.md [hunk 1]
```diff
@@ -23,12 +23,12 @@ Gardener allows you to create a Kubernetes cluster on different infrastructure p
...
 1. Get the properties of your Azure AD tenant, Subscription and Service Principal.
-    ```
-    Before you can provision and access a Kubernetes cluster on Azure, you need to add the Azure service principal, AD tenant and subscription credentials in Gardener. 
+
+    Before you can provision and access a Kubernetes cluster on Azure, you need to add the Azure service principal, AD tenant and subscription credentials in Gardener.
     Gardener needs the credentials to provision and operate the Azure infrastructure for your Kubernetes cluster.
 
-    **Ensure that the Azure service principal has the `Contributor` role within your Subscription assigned.**
-    ```
+    **Ensure that the Azure service princiapl has the actions defined [here](../../azure-permissions.md) within your Subscription assigned.
```
Comment (kon-angelo) [inline docs/tutorials/kubernetes-cluster-on-azure-with-gardener/kubernetes-cluster-on-azure-with-gardener.md:30]: typo

File: docs/tutorials/kubernetes-cluster-on-azure-with-gardener/kubernetes-cluster-on-azure-with-gardener.md [hunk 2]
```diff
@@ -23,12 +23,12 @@ Gardener allows you to create a Kubernetes cluster on different infrastructure p
...
+    If no fine-grained permission/actions required then simply the buildin `Contributer` role can be assigned.**
```
Comment (kon-angelo) [inline docs/tutorials/kubernetes-cluster-on-azure-with-gardener/kubernetes-cluster-on-azure-with-gardener.md:31]:
```suggestion
    If no fine-grained permission/actions required then simply the buildin `Contributor` role can be assigned.**
```

File: docs/tutorials/kubernetes-cluster-on-azure-with-gardener/kubernetes-cluster-on-azure-with-gardener.md (hunk 1 above)
Comment (kon-angelo) [inline docs/tutorials/kubernetes-cluster-on-azure-with-gardener/kubernetes-cluster-on-azure-with-gardener.md:30]:
```suggestion
    **Ensure that the Azure service principal has the actions defined [here](../../azure-permissions.md) within your Subscription assigned.
```

**Review by ialidzhikov — COMMENTED** [review]
Maybe you want to write in the PR description that you aim to fix https://github.com/gardener/gardener-extension-provider-azure/issues/26.

For our internal installations how we make sure that we run with the fine-grained permissions and not the `Contributor` role?
I guess would also want to adapt the help text in the gardener/dashboard - -ref https://github.com/gardener/dashboard/blob/2932891cd17dcadf185eb42899d77ab81267aac3/frontend/src/components/dialogs/SecretDialogAzure.vue#L65-L78

**Comment by dkistner** [comment]
@ialidzhikov We will provide a custom role in our Azure tenants which include those permissions and adopt our spns to use this role instead.

Thanks for the hint with the dashboard. I will provide PR to update this.

**Review by kon-angelo — APPROVED** [review]
/lgtm

**Pull request merged**